storage/tests/system.py

```diff
@@ -1061,54 +1061,51 @@ def setUpClass(cls):
     def setUp(self):
         super(TestKMSIntegration, self).setUp()
         client = kms.KeyManagementServiceClient()
+        project = Config.CLIENT.project
+        location = self.bucket.location.lower()
+        keyring_name = self.KEYRING_NAME
+        purpose = kms.enums.CryptoKey.CryptoKeyPurpose.ENCRYPT_DECRYPT
 
         # If the keyring doesn't exist create it.
-        name = client.key_ring_path(
-            Config.CLIENT.project,
-            self.bucket.location.lower(),
-            self.KEYRING_NAME)
+        keyring_path = client.key_ring_path(project, location, keyring_name)
 
         try:
-            client.get_key_ring(name)
+            client.get_key_ring(keyring_path)
         except exceptions.NotFound:
-            parent = client.location_path(
-                Config.CLIENT.project, self.bucket.location.lower())
-            client.create_key_ring(parent, self.KEYRING_NAME, {})
-
-        # Ensure this service account is marked as an owner to the test keyring
-        keyring_location_path = client.key_ring_path(
-            Config.CLIENT.project,
-            self.bucket.location.lower(),
-            self.KEYRING_NAME)
-        service_account = Config.CLIENT.get_service_account_email()
-        policy = {
-            "bindings": [
-                {
-                    "role": "roles/owner",
-                    "members": [
-                        "serviceAccount:" + service_account,
-                    ]
-                }
-            ]
-        }
-
-        client.set_iam_policy(keyring_location_path, policy)
+            parent = client.location_path(project, location)
+            client.create_key_ring(parent, keyring_name, {})
+
+            # Mark this service account as an owner of the new keyring
+            service_account = Config.CLIENT.get_service_account_email()
+            policy = {
+                "bindings": [
+                    {
+                        "role": "roles/owner",
```

Comment:
Ah I missed this line. can you reduce role to "roles/cloudkms.cryptoKeyEncrypterDecrypter" that's the only requirement for GCS to use kms keys in this kms keyring.

Reply:
e5abbcd